woocommerce/woocommerce
# Fix/28759 billing shipping addresses

### All Submissions:






### Changes proposed in this Pull Request:

In some countries the state (county) field is optional—the UK is one such example. That means we might have a saved billing address _with_ a county, and a saved shipping address _without_ (or similar). 

In these cases we should not copy the billing state/county to the shipping state/county: that action is really only appropriate if we don't already have a shipping address. Note that this problem also applies to the country field (though, it is easiest to see in relation to the state/county field).

Closes #28759.

### How to test the changes in this Pull Request:

1. Start by confirming you can see the original problem (checkout [current trunk code](https://github.com/woocommerce/woocommerce/commit/855f48d53b9a83c109891340b2b71bf9772db89f), or else the `5.1.0` tag).
    1. Create a customer account and ensure shipping is set up.
    2. Add and save billing and shipping address, supplying a county for the billing address but not for the shipping address.
    3. Now purchase a product (it's easiest to see if you buy a simple physical product).
    4. Note that the shipping address appearing on the cart page includes the billing county.
2. Now checkout this branch.
    1. Start by clearing/deleting the user session† and logging back in, otherwise the shipping address will be inherited from the first test (complete with inappropriately added state/county). 
    2. Then, simply repeat the test. The billing county is not copied over to the shipping address.

† _If you have your own snippet to kill the user session and dump the cart contents that's probably fine, or you could try using [this snippet](https://gist.github.com/barryhughes/d17134dc83dd216555115182879da5f4) (updating the user ID, and if necessary the table prefix, accordingly) then flushing the cache._

### Other information:




### Changelog entry

> Fix - If we have a non-empty shipping address then do not overwrite the state or country fields with billing address data.
